example_code/args/args.c

```diff
@@ -1,9 +1,12 @@
 #include <stdio.h>
+#include <stdlib.h>
 
 int main(int argc, char **argv)
 {
+    fprintf(stderr, "Argc: %d\n", argc);
     for (int i = 0; i < argc; i++)
     {
         printf("Arg %d: %s\n", i, argv[i]);
     }
+    exit(EXIT_SUCCESS);
```

Comment:
Is this different from a return? Or is this just testing exit?

Reply:
In our case, because the WASI exit syscall implementation just calls exit, the main difference is whether the runtime's main function gets returned to or not:
```c
int main(int argc, char* argv[]) {
    runtime_init();
    runtime_args_init(argc, argv);

    atexit(runtime_on_module_exit);

    switch_out_of_runtime();
    wasmf__start();
    /* WASI wrappes non zero status codes in exits, but silently returns on status code 0 */
    return 0;
}
```
A return would return from `wasmf__start` to the main function, but an exit skips this and runs the atexit handlers. Since we have our cleanup logic in the atexit handlers, it basically does the same thing if the modules returns or calls exit.